src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BeeListener.java

```diff
@@ -0,0 +1,68 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners;
+
+import java.util.Optional;
+
+import org.bukkit.entity.Bee;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.inventory.ItemStack;
+
+import io.github.thebusybiscuit.cscorelib2.inventory.ItemUtils;
+import io.github.thebusybiscuit.slimefun4.api.items.HashedArmorpiece;
+import io.github.thebusybiscuit.slimefun4.api.player.PlayerProfile;
+import io.github.thebusybiscuit.slimefun4.implementation.items.armor.SlimefunArmorPiece;
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+
+/**
+ * The listener for Hazmat Suit's {@link Bee} sting protection.
+ * Only applied if the whole set is worn.
+ *
+ * @author Linox
+ *
+ */
+public class BeeListener implements Listener {
+
+    public BeeListener(SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler
+    public void onDamage(EntityDamageByEntityEvent e) {
+        if (e.getDamager() instanceof Bee) {
+            if (e.getEntity() instanceof Player) {
+                Player p = (Player) e.getEntity();
+                PlayerProfile.get(p, profile -> {
+
+                    HashedArmorpiece[] armors = profile.getArmor();
+                    if (hasFullHazmat(armors)) {
+                        for (ItemStack armor : p.getInventory().getArmorContents()) {
+                            ItemUtils.damageItem(armor, 1, false);
+                        }
+                        e.setDamage(0D);
+                    }
+                });
```

Comment:
This is an asynchronous operation, so you should NOT do `e.setDamage()` here, as it will have no effect when this is delayed.
You need to the following instead:

```java
Optional<PlayerProfile> optional = PlayerProfile.find(p);
if (!optional .isPresent()) {
  PlayerProfile.request(p);
  return;
}

PlayerProfile profile = optional.get();
// Do stuff...
```

This will ensure that we don't access the PlayerProfile asynchronously, which will lead to problems.
And only execute the code if it's synced.